File: AnalyticsProject/ViewModels/ComparedStatsVM.cs
using AnalyticsProject.DataModels;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace AnalyticsProject.ViewModels
{
    public class ComparedStatsVM
    {
        public Guid Id { get; set; }
        public int averageFollowerIncrease { get; set; }
        public int averageLikesIncrease { get; set; }
        public int averageRetweetsIncrease { get; set; }
        public int averageCommentsIncrease { get; set; }
        public List<DateTime> bestPostTime { get; set; }
        public List<string> mostCommonEffectiveWords { get; set; }
        public List<FacebookDbVM> top5Posts { get; set; }
        public List<EventsVM> EventsUsed { get; set; }
        
        public ComparedStatsVM() {
        }
    }
}

File: AnalyticsProjectApp/AnalyticsProjectApp/src/app/models/comparedStatsVM.ts
import { eventsVM } from "./eventsVM";

export class comparedStatsVM  {
    public Id: any;
    public averageFollowerIncrease!: number;
    public averageLikesIncrease!: number;
    public averageRetweetIncrease!: number;
    public averageCommentsIncrease!: number;
    public bestPostTime!: Array<Date>;
    public mostCommonEffectiveWord!: Array<string>;
    public top5Posts!: Array<any>;
    public EventsUsed!: Array<eventsVM>;
  
    constructor(data?: any) {
      const self = this;

      if (!data) {
        data = {};
      }

      Object.assign(self, data);
    }
  }